### PR TITLE
Initial pass on some improvements

### PR DESCRIPTION
**Pull request by ameihm0912**
A bunch of changes in here, mostly reliability/cleanup/testing related

This PR is focused on the initial audit configuration and resolves intermittent failures running the tests.

- Change makefile so make test works on branches/forks and rebuilds constants if needed
- Reduce complexity of various netlink related functions
- Remove use of unsafe package in areas performance is not important (e.g., configuration); unsafe.Pointer related logic has been changed to use binary.Write
- Add alignment to outgoing netlink messages
- Modify netlink message parsing to avoid unsafe.Pointer copies and honor NLM_F_MULTI
- Remove code duplication in netlink message reception and use a single function for responses and for consuming kernel ACKs; add missing kernel ACK reception where needed
- Change rules loading so it uses a standard JSON struct instead of an anonymous struct / interface{}; this was causing problems iterating over map values as the order of audit rules was not consistent between rule loads
- Various other style changes to simplify the code

### CODE REVIEW

**Comment by ameihm0912** [comment]
@arunk-s @gdestuynder r?

**Comment by arunk-s** [comment]
@ameihm0912 Nice! Give me a couple of days to review :)

File: rules.go [hunk 1]
```diff
@@ -521,159 +524,120 @@ It expects the config in a json formatted string of following format:
...
+			return err
+		}
+		fpd := fieldPairData{
+			fieldval:     x.Key,
+			opval:        AUDIT_EQUAL,
+			fieldname:    "key",
+			flags:        AUDIT_FILTER_UNSET,
+			syscallAdded: true,
+		}
+		err = auditRuleFieldPairData(&ruleData, &fpd)
```
Comment (ameihm0912) [inline rules.go]: Ahhh yes, I see this value is optional. So I am assuming what we would want to do here, is only apply the field pair data if key is set; if not we just add the rule without it.

Comment (arunk-s) [inline rules.go]: Yeah, if `key` param is given, only then we set the field pair data for it.

Comment (ameihm0912) [inline rules.go]: @arunk-s thanks, fixed this up

**Comment by gdestuynder** [comment]
r+ ;-)

**Pull request merged**